viz/cmd/prune.go

```diff
@@ -0,0 +1,58 @@
+package cmd
+
+import (
+	"strings"
+	"time"
+
+	pkgCmd "github.com/linkerd/linkerd2/pkg/cmd"
+	"github.com/linkerd/linkerd2/pkg/flags"
+	"github.com/linkerd/linkerd2/pkg/healthcheck"
+	"github.com/spf13/cobra"
+	"helm.sh/helm/v3/pkg/cli/values"
+)
+
+func newCmdPrune() *cobra.Command {
+	var ha bool
+	var cniEnabled bool
+	var wait time.Duration
+	var options values.Options
+
+	cmd := &cobra.Command{
+		Use:   "prune [flags]",
+		Args:  cobra.NoArgs,
+		Short: "Output extraneous Kubernetes resources in the linkerd-viz extension",
+		Long:  `Output extraneous Kubernetes resources in the linkerd-viz extension.`,
+		Example: `  # Prune extraneous resources.
+  linkerd viz prune | kubectl delete -f -
+  `,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			hc := healthcheck.NewWithCoreChecks(&healthcheck.Options{
+				ControlPlaneNamespace: controlPlaneNamespace,
+				KubeConfig:            kubeconfigPath,
+				KubeContext:           kubeContext,
+				Impersonate:           impersonate,
+				ImpersonateGroup:      impersonateGroup,
+				APIAddr:               apiAddr,
+				RetryDeadline:         time.Now().Add(wait),
+			})
+			hc.RunWithExitOnError()
+			cniEnabled = hc.CNIEnabled
+
+			manifests := strings.Builder{}
+
+			err := install(&manifests, options, ha, cniEnabled)
+			if err != nil {
+				return err
+			}
+
+			return pkgCmd.Prune(cmd.Context(), hc.KubeAPIClient(), manifests.String(), "linkerd.io/extension=viz")
```

Comment:
Nit (requires adding import `labels "github.com/linkerd/linkerd2/pkg/k8s"`, and also applies to the other extensions):
```suggestion
                         label := fmt.Sprintf("%s=%s", labels.LinkerdExtensionLabel, healthcheck.VizExtensionName)
			return pkgCmd.Prune(cmd.Context(), hc.KubeAPIClient(), manifests.String(), label)
```